sdcm/remote/libssh2_client/__init__.py

```diff
@@ -440,25 +440,31 @@ def _process_output(  # pylint: disable=too-many-arguments, too-many-branches
             if perf_counter() > end_time:
                 reader.stop()
                 return False
-            if stdout_stream is not None:
+
+            data_processed = False
+            if stdout_stream is not None and reader.stdout.qsize():
                 try:
                     stdout = reader.stdout.get(timeout=timeout_read_data_chunk)
                     data = stdout.decode(encoding, errors='ignore') + '\n'
                     stdout_stream.write(data)
                     for watcher in watchers:
                         watcher.submit_line(data)
+                    data_processed = True
                 except Exception:  # pylint: disable=broad-except  # noqa: BLE001
                     pass
-            if stderr_stream is not None:
+            if stderr_stream is not None and reader.stderr.qsize():
                 if reader.stderr.qsize():
                     try:
                         stderr = reader.stderr.get(timeout=timeout_read_data_chunk)
                         data = stderr.decode(encoding) + '\n'
                         stderr_stream.write(data)
                         for watcher in watchers:
                             watcher.submit_line(data)
+                        data_processed = True
                     except Exception:  # pylint: disable=broad-except  # noqa: BLE001
                         pass
+            if not data_processed:  # prevent unbounded loop in case no data on the streams
+                sleep(timeout_read_data_chunk)
```

Comment:
Are we sure that the `timeout_read_data_chunk` is integer/float here?

Reply:
no, it can be `None`. I added proper condition.